jwst/master_background/master_background_step.py

```diff
@@ -114,15 +118,18 @@ def process(self, input):
                         background_data,
                         exptime_key='exposure_time',
                         background=True,
-                    )
+                        )
 
                     result = datamodels.ModelContainer()
                     result.update(input_data)
                     for model in input_data:
                         background_2d = expand_to_2d(model, master_background)
                         result.append(subtract_2d_background(model, background_2d))
 
-                # Skip step for case with no container and no user background
+                # Record name of user-supplied master background spectrum
+                if isinstance(result, datamodels.ModelContainer):
+                    for model in result:
+                        model.meta.background.master_background_file = basename(self.user_background)
```

Comment:
I don't think this block should be here?

Reply:
I messed up the rebase and conflicts. Dang it . Thanks for catching it.